### PR TITLE
feat(config): add optional EnigmAgent LLM credential override

### LINKED ISSUE #769: feature: EnigmAgent credential injection for production secret management
### Problem

OpenSRE currently resolves LLM API keys and cloud credentials through direct `os.getenv()` calls or dotenv files (e.g. `app/config.py:resolve_llm_api_key()`). In production SRE environments this has known shortcomings:

- Secrets in `.env` files can leak into container images or logs
- No rotation without a full redeploy
- No audit trail for which investigation used which credential
- Multi-tenant deployments (different teams, different cloud accounts) need per-run credential scoping

### Proposal

Integrate **[EnigmAgent](https://github.com/Agnuxo1/EnigmAgent)** as an optional credential-injection layer.

EnigmAgent is a zero-trust credential broker designed for agentic LLM workflows. At investigation start-time it:
1. Authenticates the calling agent identity (via agent ID + environment fingerprint)
2. Retrieves the minimum-privilege credentials for the target provider from a vault backend (HashiCorp Vault, AWS Secrets Manager, or its own lightweight store)
3. Returns scoped, time-limited credentials injected directly into the `LLMSettings` and cloud-provider clients — never persisted to disk

### Integration surface

The smallest viable integration is a single hook in `LLMSettings.from_env()`:

```python
# app/config.py

if os.getenv("ENIGM_AGENT_URL"):
    from enigmagent import EnigmClient
    creds = EnigmClient(os.getenv("ENIGM_AGENT_URL")).resolve(scope="opensre")
    # creds maps env-var names → values; merge before standard getenv resolution
    os.environ.update(creds)
```

The block is a no-op when `ENIGM_AGENT_URL` is unset, so existing deployments are completely unaffected.

### Benefits for OpenSRE

| Concern | Current | With EnigmAgent |
|---|---|---|
| Secret rotation | Redeploy required | Hot-rotated; next investigation gets fresh token |
| Audit trail | None | Timestamped credential request per investigation run |
| Multi-tenant | Manual env management | Per-agent scoped credentials via identity claim |
| Leaked `.env` | High risk | Never touches filesystem |

### Suggested scope

- [ ] Add optional `ENIGM_AGENT_URL` env var check in `LLMSettings.from_env()`
- [ ] Docs: add "Secret Management" section to README with EnigmAgent setup example
- [ ] (Optional) Makefile target: `make setup-secrets` that bootstraps a local EnigmAgent instance

Happy to draft a PR if there is interest. Feedback on whether this fits the project's operational model would also be very welcome.

### PR DESCRIPTION
**Pull request by MestreY0d4-Uninter**
Fixes #769

## Type of Change
- [ ] Bug fix
- [x] Feature
- [ ] Breaking change
- [ ] Docs

## What Changed
This PR implements the minimal first slice discussed in issue #769: an optional EnigmAgent-backed credential override for `LLMSettings.from_env()`.

Scope intentionally kept small and reviewable:
- add optional `ENIGM_AGENT_URL` handling inside `LLMSettings.from_env()`
- keep the path a complete no-op when `ENIGM_AGENT_URL` is unset
- add `enigmagent` as an optional dependency in `pyproject.toml`
- add two focused tests in `tests/test_config.py`

### Implementation details
- Added a small helper in `app/config.py` that:
  - parses `ENIGM_AGENT_URL`
  - lazily imports the optional `enigmagent` dependency
  - uses EnigmAgent's Python SDK client to resolve the selected provider API key
- The resolved value is merged into the config payload passed to `LLMSettings.model_validate()`.
- This does **not** mutate `os.environ`; the override is local to config construction.
- If `ENIGM_AGENT_URL` is absent, existing behavior is unchanged.

### SDK note
The issue body used pseudocode with `EnigmClient(...).resolve(scope=...)`, but the current Python SDK exposes `VaultClient` / `resolve(...)`. This PR follows the current SDK surface while preserving the requested minimal integration shape.

## Testing
Ran local validation in the worktree:
- ✅ `uv run pytest tests/test_config.py -q`
- ✅ `uv run make lint`
- ✅ `uv run make typecheck`
- ✅ `uv run make test-cov`

### Test coverage added
- no-op when `ENIGM_AGENT_URL` is absent
- override path when `ENIGM_AGENT_URL` is present and EnigmAgent returns a provider API key

## Impact Analysis
- **Backward compatible:** Yes
- **Default behavior changed when `ENIGM_AGENT_URL` is unset:** No
- **Environment mutation:** None
- **Scope:** limited to config construction + tests + optional dependency entry

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

I intentionally kept this first PR limited to the smallest integration surface suggested in the issue discussion.

I considered two implementation styles:
1. mutate `os.environ` before normal resolution
2. resolve EnigmAgent values and merge them only into the `model_validate()` input

I chose option 2 because it keeps `LLMSettings.from_env()` free of global side effects and makes the behavior easier to test.

The change only affects the selected provider API-key resolution path when `ENIGM_AGENT_URL` is configured. Otherwise, `resolve_llm_api_key()` continues to behave exactly as before.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] I can explain the purpose of every function, class, and logic block I added
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added corresponding tests
- [x] My code follows the project's style guidelines and conventions

### CODE REVIEW

**Comment by MestreY0d4-Uninter** [comment]
Updated this PR on top of current `main` and addressed the Greptile findings:

- restricted `ENIGM_AGENT_URL` to plain HTTP loopback hosts only (`localhost`, `127.0.0.1`, `::1`), so resolved credentials cannot be fetched over plaintext from remote hosts;
- added an autouse test fixture that clears `ENIGM_AGENT_URL` for `tests/test_config.py`, preventing ambient environment leakage into existing `from_env()` tests;
- kept the EnigmAgent dependency optional and lazily imported only when `ENIGM_AGENT_URL` is configured.

Local validation:
- `python -m pytest -q tests/test_config.py` — 11 passed
- `make lint` — passed
- `make format-check` — passed
- `make typecheck` — passed
- `make test-cov` — passed on rerun: 3158 passed, 2 skipped, 1 warning

GitHub checks for the updated head are now green: quality, typecheck, test, and CodeQL passed.

File: app/config.py [hunk 1]
```diff
@@ -124,6 +125,55 @@ def get_environment() -> Environment:
...
+        host=parsed.hostname,
+        port=parsed.port or 3737,
+        timeout=5.0,
+        origin="http://localhost",
+    )
+
+    try:
+        value = client.resolve(env_var, origin="http://localhost").strip()
+    except Exception as exc:
+        if getattr(exc, "code", None) in _ENIGMAGENT_IGNORED_ERROR_CODES:
```
Comment (muddlebee) [inline app/config.py:171]: **Required:** This ignored-error-codes path is not covered by tests. If a future refactor changes exception handling (e.g., switching from getattr to a typed exception), this silent fallback could regress. Please add a test that injects a fake VaultClient raising an exception with one of these codes and asserts the fallback to resolve_llm_api_key still works.

File: app/config.py [hunk 2]
```diff
@@ -124,6 +125,55 @@ def get_environment() -> Environment:
     "codex",
 ]
 
+_ENIGMAGENT_IGNORED_ERROR_CODES = frozenset({"not_found", "no_domain_binding", "domain_mismatch"})
+_PROVIDER_API_KEY_ENV_VARS = {
+    "anthropic": "ANTHROPIC_API_KEY",
+    "openai": "OPENAI_API_KEY",
```
Comment (muddlebee) [inline app/config.py:131]: **Consider:** This mapping duplicates the provider→env-var mapping already present in _require_api_key_for_selected_provider (lines ~187-204).

**Review by muddlebee — CHANGES_REQUESTED** [review]
**Action items:**
1. **Required:** Add a test covering the ignored EnigmAgent error-code fallback (e.g., `test_llm_settings_from_env_enigmagent_ignored_errors_fallback`).
2. **Optional:** Consider DRYing up the provider→env-var mapping so `_PROVIDER_API_KEY_ENV_VARS` and `_require_api_key_for_selected_provider` share a single source of truth.

Once the required test is in, this is ready to merge.

**Comment by MestreY0d4-Uninter** [comment]
Updated in `e73628b06446daeef9e00a11a88dfbeff1659d15`.

Addressed the required review item:
- Added `test_llm_settings_from_env_enigmagent_ignored_errors_fallback`, which injects a fake `VaultClient` raising an ignored EnigmAgent error code and verifies `LLMSettings.from_env()` falls back to the normal API-key resolver.

Also addressed the optional note from follow-up review:
- Added a comment documenting that the current EnigmAgent flow intentionally limits plain HTTP to loopback, and that the validator can be relaxed if the SDK supports TLS later.

Notes:
- The nested `resolve_api_key` helper already has `-> str` on the current branch.
- I left the provider mapping DRY-up out of this update because it was marked optional and would broaden the PR beyond the required merge blocker.

Validation:
- `python -m pytest -q tests/test_config.py` — 12 passed
- `make lint` — passed
- `make format-check` — passed
- `make typecheck` — passed
- `make test-cov` — 3159 passed, 2 skipped, 1 warning
- GitHub checks on the updated head are green: quality, Analyze (python), typecheck, test, and CodeQL passed.

**Comment by muddlebee** [comment]
@MestreY0d4-Uninter looks good only. can you share a video demo where you have tested with those API keys and changes?

**Comment by MestreY0d4-Uninter** [comment]
Thanks! I recorded a real terminal-session demo for the current PR head.

The terminal input was automated for repeatability, but the commands executed in a live shell. The disposable Gemini key is read from a local restricted file and is never printed raw. The run starts a real EnigmAgent REST server, resolves `GEMINI_API_KEY` through EnigmAgent, exercises OpenSRE `LLMSettings.from_env()`, validates Gemini over HTTPS, and passes `tests/test_config.py`.
https://github.com/user-attachments/assets/f01a79fb-adae-4037-a8cf-6d05936ee6ce

**Comment by MestreY0d4-Uninter** [comment]
Hi @muddlebee, the required fallback test was added in e73628b and all CI checks are green. I also shared the requested masked demo with API-key evidence. Could you please re-review when you have a chance? Thanks!

**Comment by muddlebee** [comment]
hey @MestreY0d4-Uninter thank you for the efforts. but this needs more discussion with the community and the package https://github.com/Agnuxo1/EnigmAgent needs to be reviewed for security and stability.
